.github/workflows/cd-docs.yml

```diff
@@ -0,0 +1,51 @@
+name: deploy-docs
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+permissions:
+  contents: write
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: 'pip'
+          cache-dependency-path: |
+            tensorflow_serving/tools/pip_package/setup.py
+
+      - name: Save time for cache for mkdocs
+        run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV
+
+      - name: Caching
+        uses: actions/cache@v4
+        with:
+          key: mkdocs-material-${{ env.cache_id }}
+          path: .cache
+          restore-keys: |
+            mkdocs-material-
+
+      - name: Install Dependencies
+        run: |
+          sudo apt install -y doxygen graphviz
+          pip install -r tensorflow_serving/tools/pip_package/requirements-docs.txt
+
+      - name: Deploy to GitHub Pages
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+          mkdocs gh-deploy --force
+        if: (github.event_name != 'pull_request')
+
+      - name: Build docs to check for errors
+        run: |
+          mkdocs build
+        if: (github.event_name == 'pull_request')
```

Comment:
Does this need to be before the `gh-deploy` step?

Reply:
No. [`gh-deploy` builds the docs](https://www.mkdocs.org/user-guide/deploying-your-docs/#project-pages).